receiver/snmpreceiver/scraper.go

```diff
@@ -0,0 +1,559 @@
+	// Find all attribute column OIDs
+	indexedAttributeOIDs := []string{}
+	for name, attributeCfg := range s.cfg.Attributes {
+		if attributeCfg.OID != "" {
+			// Data is returned by the client with '.' prefix on the OIDs.
+			// Making sure the prefix exists here in the configs so we can match it up with returned data later
+			if !strings.HasPrefix(attributeCfg.OID, ".") {
+				attributeCfg.OID = "." + attributeCfg.OID
+				s.cfg.Attributes[name] = attributeCfg
+			}
+			indexedAttributeOIDs = append(indexedAttributeOIDs, attributeCfg.OID)
+		}
+	}
+
+	// If no oid attribute configs, nothing else to do
+	if len(indexedAttributeOIDs) == 0 {
+		return nil
+	}
+
+	// Retrieve SNMP attribute indexed data and store for later use
+	err = s.client.GetIndexedData(
+		indexedAttributeOIDs,
+		indexedDataToAttribute(indexedAttributeMapByOID),
+	)
+
+	return err
+}
+
+// scrapeIndexedResourceAttributes retrieves all SNMP data from resource attribute config column OIDs and
+// stores the returned indexed data for later use by metrics
+func (s *snmpScraper) scrapeIndexedResourceAttributes(indexedResourceAttributeMapByOID map[string]map[string]string) (err error) {
+	// Find all resource attribute column OIDs
+	indexedResourceAttributeOIDs := []string{}
+	for name, resourceAttributeCfg := range s.cfg.ResourceAttributes {
+		if resourceAttributeCfg.OID != "" {
+			// Data is returned by the client with '.' prefix on the OIDs.
+			// Making sure the prefix exists here in the configs so we can match it up with returned data later
+			if !strings.HasPrefix(resourceAttributeCfg.OID, ".") {
+				resourceAttributeCfg.OID = "." + resourceAttributeCfg.OID
+				s.cfg.ResourceAttributes[name] = resourceAttributeCfg
+			}
+			indexedResourceAttributeOIDs = append(indexedResourceAttributeOIDs, resourceAttributeCfg.OID)
+		}
+	}
+
+	// If no OID resource attribute configs, nothing else to do
+	if len(indexedResourceAttributeOIDs) == 0 {
+		return nil
+	}
+
+	// Retrieve resource attribute indexed data and store for later use
+	err = s.client.GetIndexedData(
+		indexedResourceAttributeOIDs,
+		indexedDataToAttribute(indexedResourceAttributeMapByOID),
+	)
+
+	return err
+}
+
+// indexedDataToAttribute provides a function which will take one piece of column OID SNMP indexed data and
+// stores it in a nested map for later use (keyed by both [resource] attribute config column OID and
+// indexed OID value index)
+func indexedDataToAttribute(
+	indexedAttributeMapByOID map[string]map[string]string,
+) processFunc {
+	// This returns a processFunc because this is what the client's GetIndexedData method requires
+	return func(data snmpData) error {
+		// Get the string value of the SNMP data for the [resource] attribute value
+		var stringValue string
+		switch data.valueType {
+		case NotSupported:
+			return fmt.Errorf(errMsgIndexedAttributesBadValueType, data.oid, data.parentOID)
+		case String:
+			stringValue = data.value.(string)
```

Comment:
When we do these casts, we should try to avoid a panic (I think that's what happens if it can't cast?), e.g.:
```go
stringValue, ok = data.value.(string)
if !ok {
    return fmt.Errorf("...")
}
```

Reply:
Actually, this is probably overkill, the method that sets up these values looks pretty straght-forward: https://github.com/tamir-michaeli/opentelemetry-collector-contrib/blob/653774970b3ebd7b4c48135f1b666cb9adf18ef0/receiver/snmpreceiver/client.go#L387

Reply:
@BinaryFissionGames i'll at least add a comment to explain why i'm not checking to make it more clear